### PR TITLE
feat(docs-infra): update CLI templates to match changes with new parser and folder structure

### PR DESCRIPTION
**Pull request by alan-agius4**
The underlying parser that the CLI uses changed which caused minor changes in the generated JSON helps especially for subcommands.

The folder structure of the Angular CLI repo also changed slightly.

More context: https://github.com/angular/angular-cli/pull/22778

### CODE REVIEW

**Comment by alan-agius4** [comment]
CI will fail until we update the CLI SHA. Can I get a preemptive review?

File: aio/tools/transforms/cli-docs-package/processors/processCliCommands.js [hunk 1]
```diff
@@ -29,6 +43,31 @@ module.exports = function processCliCommands(createDocMessage) {
...
+  const usage = parentCommand
+    ? // `ng generate <schematic>` -> `ng generate app-shell`
+      parentCommand.command.split(/ [<|[]/, 1)[0] + ' ' + command.command
+    : // `ng build [project]`
+      command.command;
+
+  // Handle required and optional command line args.
+  // Wrap them in a <var> element.
+  const commandUsageHtmlString = usage
+    .replace(/>/g, '&gt;&gt;/var>')
```
Comment (gkalpak) [inline aio/tools/transforms/cli-docs-package/processors/processCliCommands.js:56]: Shouldn't this be:

```suggestion
    .replace(/>/g, '&gt;/var>&gt;')
```

File: aio/tools/transforms/templates/cli/lib/cli.html [hunk 2]
```diff
@@ -21,17 +20,9 @@
   <tbody>
   {% for option in arguments %}
   <tr class="cli-option">
-    <td><code class="cli-option-syntax no-auto-link">&lt;<var>{$ option.name $}</var>&gt;</code></td>
+    <td><code class="cli-option-syntax no-auto-link"><var>{$ option.name $}</var></code></td>
```
Comment (gkalpak) [inline aio/tools/transforms/templates/cli/lib/cli.html:23]: OOC, why this change?

File: aio/tools/transforms/templates/cli/cli-command.template.html [hunk 3]
```diff
@@ -2,7 +2,6 @@
 {% import "../lib/githubLinks.html" as github -%}
 
 <article>
-  {$ github.githubLinks(doc, cliVersionInfo) $}
```
Comment (gkalpak) [inline aio/tools/transforms/templates/cli/cli-command.template.html:5]: OOC, why this change?

**Review by gkalpak — COMMENTED** [review]
Preemptively looks reasonable :smiley:

File: aio/tools/transforms/templates/cli/lib/cli.html (hunk 2 above)
Comment (alan-agius4) [reply aio/tools/transforms/templates/cli/lib/cli.html:23]: Because `<…>`, means it’s a required positional argument which in this case it can be wrong and should be displayed as part of the description as `<>` or  `[]` will be displayed in the usage.

File: aio/tools/transforms/templates/cli/cli-command.template.html (hunk 3 above)
Comment (alan-agius4) [reply aio/tools/transforms/templates/cli/cli-command.template.html:5]: The JSON schemas no longer exist.

File: aio/tools/transforms/cli-docs-package/processors/processCliCommands.js (hunk 1 above)
Comment (alan-agius4) [reply aio/tools/transforms/cli-docs-package/processors/processCliCommands.js:56]: You are right

File: aio/tools/transforms/templates/cli/cli-command.template.html (hunk 3 above)
Comment (gkalpak) [reply aio/tools/transforms/templates/cli/cli-command.template.html:5]: We should still point people to the source imo. Where will the docs live in the future?

Comment (alan-agius4) [reply aio/tools/transforms/templates/cli/cli-command.template.html:5]: The schemas has been replaced with a more declarative approach 

Example:
https://github.com/angular/angular-cli/blob/a4b89722633689541565be536e0b6560fb885ff2/packages/angular/cli/src/commands/config/cli.ts

https://github.com/angular/angular-cli/blob/a4b89722633689541565be536e0b6560fb885ff2/packages/angular/cli/src/commands/run/cli.ts

Comment (gkalpak) [reply aio/tools/transforms/templates/cli/cli-command.template.html:5]: So, could we fix the [realProjectRelativePath][1] to point to the new locations instead and keep the GitHub links?

[1]: https://github.com/angular/angular/blob/dc6094acdcd9bfd8409c6cb0aadd4638394f5df1/aio/tools/transforms/cli-docs-package/readers/cli-command.js#L18

Comment (alan-agius4) [reply aio/tools/transforms/templates/cli/cli-command.template.html:5]: We could, although in most of the cases the options are a combinations of multiple files. This was also the case previously which wasn't handled as the link pointed to the build schema ex (https://github.com/angular/angular-cli/blob/master/packages/angular/cli/commands/build.json).

Example for `ng build`
- https://github.com/angular/angular-cli/blob/a4b89722633689541565be536e0b6560fb885ff2/packages/angular/cli/src/commands/build/cli.ts
-https://github.com/angular/angular-cli/blob/a4b89722633689541565be536e0b6560fb885ff2/packages/angular/cli/src/command-builder/architect-command-module.ts#L31-L44
- https://github.com/angular/angular-cli/blob/a4b89722633689541565be536e0b6560fb885ff2/packages/angular_devkit/build_angular/src/builders/browser/schema.json

In general, I don't see the edit link in the here to add much value considering the most likely the file that will be linked will be the incorrect one. As a matter of fact, I don't recall any PRs were opened from AIO.

Comment (gkalpak) [reply aio/tools/transforms/templates/cli/cli-command.template.html:5]: Fair enough. If you don't think it adds value, let's omit the links.

**Comment by alan-agius4** [comment]
@gkalpak and @josephperrott kindly take a look.

File: aio/tools/transforms/templates/cli/cli-command.template.html [hunk 4]
```diff
@@ -2,15 +2,13 @@
...
   {% include 'include/cli-breadcrumb.html' %}
   {% include 'include/cli-header.html' %}
 
   <aio-toc class="embedded"></aio-toc>
 
   <div class="cli-body">
     {$ doc.shortDescription | marked $}
-    {$ doc.description | marked $}
```
Comment (gkalpak) [inline aio/tools/transforms/templates/cli/cli-command.template.html:13]: OOC, why remove this? Won't these docs have descriptions any more?

File: aio/tools/transforms/templates/cli/lib/cli.html [hunk 5]
```diff
@@ -71,7 +62,7 @@
         {% if option.aliases.length %}<p><span class="cli-aliases">Aliases:</span> {% for alias in option.aliases %}{$ renderOptionName(alias) $}{% if not loop.last %}, {% endif %}{% endfor %}</p>{% endif %}
       </td>
       <td><code class="no-auto-link">{$ renderOptionValues(option.type, option.enum) $}</td>
-      <td>{% if option.default !== undefined %}<code class="no-auto-link">{$ option.default $}</code>{% endif %}</td>
+      <td>{% if option.default !== undefined %}{% if option.default !== '' %}<code class="no-auto-link">{$ option.default $}</code>{% endif %}{% endif %}</td>
```
Comment (gkalpak) [inline aio/tools/transforms/templates/cli/lib/cli.html:65]: Could this be:

```suggestion
      <td>{% if option.default !== undefined and option.default !== '' %}<code class="no-auto-link">{$ option.default $}</code>{% endif %}</td>
```

File: aio/tools/transforms/cli-docs-package/processors/processCliCommands.js [hunk 6]
```diff
@@ -19,6 +19,20 @@ module.exports = function processCliCommands(createDocMessage) {
           // Recursively process the options
           const optionKeywords = new Set();
           processOptions(doc, doc.options, optionKeywords);
+
+          doc.usages = generateUsages(doc);
+
+          // Recurse if there are subcommands
```
Comment (josephperrott) [inline aio/tools/transforms/cli-docs-package/processors/processCliCommands.js:25]: So this only goes one level down correct, do we need to fully recurse further in?

**Review by gkalpak — APPROVED** [review]
A couple of questions. Otherwise lgtm :+1:

**Review by josephperrott — APPROVED** [review]
LGTM

We should definitely have @gkalpak review as well though as he has more historical context and understanding.

File: aio/tools/transforms/templates/cli/cli-command.template.html (hunk 4 above)
Comment (alan-agius4) [reply aio/tools/transforms/templates/cli/cli-command.template.html:13]: description has been replaced with shortDescription.

File: aio/tools/transforms/cli-docs-package/processors/processCliCommands.js (hunk 6 above)
Comment (alan-agius4) [reply aio/tools/transforms/cli-docs-package/processors/processCliCommands.js:25]: Not currently, since we only have 1 level subcommands in the Angular CLI

File: aio/tools/transforms/templates/cli/lib/cli.html (hunk 5 above)
Comment (alan-agius4) [reply aio/tools/transforms/templates/cli/lib/cli.html:65]: 👍

**Comment by atscott** [comment]
This PR was merged into the repository by commit 7f043c98b7e6d277eec060361063adee5f12a7b6.

File: aio/tools/transforms/templates/cli/cli-command.template.html (hunk 4 above)
Comment (gkalpak) [reply aio/tools/transforms/templates/cli/cli-command.template.html:13]: 🤷‍♂️

Comment (alan-agius4) [reply aio/tools/transforms/templates/cli/cli-command.template.html:13]: 😂
